libbeat/outputs/elasticsearch/client.go

```diff
@@ -82,6 +84,12 @@ func NewClient(
 		index:  index,
 		params: params,
 	}
+
+	client.Connection.onConnectCallback = func() {
```

Comment:
I gave this a try, but to my taste the code was becoming less readable. Part of the reason is that I have two types of callbacks, one with a client parameter, one without. That's because in Connection I don't have a pointer to the Client, so I used another closure to set it.

Perhaps an easier approach would be just have a pointer to the client in the connection.